pandas/tests/frame/test_asof.py

```diff
@@ -75,9 +74,15 @@ def test_missing(self):
         result = df.asof('1989-12-31')
 
         expected = Series(index=['A', 'B'], name=Timestamp('1989-12-31'))
-        assert_series_equal(result, expected)
+        tm.assert_series_equal(result, expected)
 
         result = df.asof(to_datetime(['1989-12-31']))
         expected = DataFrame(index=to_datetime(['1989-12-31']),
                              columns=['A', 'B'], dtype='float64')
-        assert_frame_equal(result, expected)
+        tm.assert_frame_equal(result, expected)
+
+    def test_all_nans(self):
+        # series is all nans
+        result = DataFrame([np.nan]).asof([0])
```

Comment:
Indeed, and also, when you have a DataFrame with multiple columns, those columns should be preserved in the result